lib/matplotlib/backends/backend_agg.py

```diff
@@ -208,10 +208,13 @@ def draw_text(self, gc, x, y, s, prop, angle, ismath=False, mtext=None):
         xo, yo = font.get_bitmap_offset()
         xo /= 64.0
         yo /= 64.0
+        h = sqrt(xo**2 + yo**2)
         xd = d * sin(radians(angle))
         yd = d * cos(radians(angle))
+        xo = h * cos(radians(angle))
+        yo = h * sin(radians(angle))
         x = round(x + xo + xd)
-        y = round(y + yo + yd)
+        y = round(y + yo - yd)
```

Comment:
much like the trig above, I think this will also need a quick proof in the comments.

Reply:
I don't think I have a proof for this at the moment except that in the draw_mathtext function the y is calculated in a similar manner and by subtracting the y offset we get a satisfactory result.